model/metadata/test_approved_system/transform_ignored_hostname.approved.json

```diff
@@ -0,0 +1,3 @@
+{
+    "name": "custom hostname"
```

Comment:
why doesn't this get `hostname` set to the `hostname` provided?  This case shouldn't happen in practice right?

Reply:
The intake API accepts 3 fields relate to hostname: `hostname`, `detected_hostname` and `configured_hostname`. 

`hostname` is deprecated, while the other two fields are newly introduced. Agent devs are expected to not use `hostname` any more once they implement the new fields. Therefore, as soon as at least one of the new fields is sent, the deprecated one is ignored, as it is not supposed to be sent anymore. See also https://github.com/elastic/apm-server/issues/2502#issuecomment-519487367